core/internal/cltest/event_websocket_server.go

```diff
@@ -101,9 +103,18 @@ func (wss *EventWebSocketServer) handler(w http.ResponseWriter, r *http.Request)
 			wss.t.Fatal("EventWebSocketServer ReadMessage: ", err)
 		}
 
-		select {
-		case wss.Received <- string(payload):
-		default:
+		if messageType == websocket.TextMessage {
+			select {
```

Comment:
Hmm, this looks a little suspicious, what are you trying to do here?

Reply:
Forward the message received on the websocket to the correct channel so that it can be asserted in a test. Wasn't sure of the best way to do this.

Another approach I considered was to keep 1 received channel but change the type `chan string` -> `chan []byte` and do the type conversion in the test, but that makes each test a bit more ugly.

Reply:
Forward the message received on the websocket to the correct channel so that it can be asserted in a test. Wasn't sure of the best way to do this.

Another approach I considered was to keep 1 received channel but change the type `chan string` -> `chan []byte` and do the type conversion in the test, but that makes each test a bit more ugly.

Reply:
Oh I'm referring to the select statement here, the way you've written it, I believe it'll just drop the message if the channel is full.